scripts/openzfs-merge.sh

```diff
@@ -0,0 +1,231 @@
+#	sh autogen.sh
+#	./configure --enable-debug
+#
+# mandatory git settings:
+#	[merge]
+#   	renameLimit = 999999
+#	[user]
+#       email = mail@gmelikov.ru
+#       name = George Melikov
+#
+# Copyright (c) 2016 George Melikov. All rights reserved.
+#
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+REPOSITORY_PATH='.'
+TMP_FILE='/tmp/gittmpmessage.txt'
+
+# list with potential OpenZFS commits
+UNPORTED_COMMITS_FILE=$SCRIPTDIR'/hashes.txt'
+
+# Next files will generate automatically
+# 	list with commits git can't merge automatically
+EXCEPTIONS_GIT=$SCRIPTDIR'/exceptions.txt'
+#	list with commits which can't be compiled without errors
+EXCEPTIONS_COMPILE=$SCRIPTDIR'/uncompiled.txt'
+#	list with commits which has cstyle error
+EXCEPTIONS_CSTYLE=$SCRIPTDIR'/unsctyled.txt'
+#	list with merged
+EXCEPTIONS_MERGED=$SCRIPTDIR'/merged.txt'
+
+LGREEN='\033[1;32m'		#  ${LGREEN}
+NORMAL='\033[0m'		#  ${NORMAL} 
+COUNT_MERGED=0
+LIST_MERGED=
+COUNT_COMPILE=0
+LIST_COMPILE=
+COUNT_CSTYLE=0
+LIST_CSTYLE=
+
+usage() {
+cat << EOF
+USAGE:
+$0 [-h] [-d directory] [-i commits.txt] [-c commit hash] [-g commit hash]
+
+DESCRIPTION:
+	Auto merge OpenZFS commits to ZFS on Linux git
+	repositories.
+	Result - git branch with name 'autoport-oz#issue'
+
+OPTIONS:
+	-h		Show this message
+	-d directory	Git repo with openzfs and zfsonlinux remotes
+	-i commits.txt	File with OpenZFS commit hashes to merge
+	(one hash per row)
+	-i commit hash	Prepare branch and try to merge this commit by hash,
+	leaves branch for manual merge if not successfull
+	-g commit hash	Generate commit description for existing commit in
+	branch generated by -i parameter
+
+EOF
+}
+
+clean_unmerged() {
+	git cherry-pick --abort
+	git checkout master
+	git branch -D "autoport-oz$OPENZFS_ISSUE"
+	rm "$TMP_FILE"
+}
+
+prepare_git() {
+	cd "$REPOSITORY_PATH"
+	rm "$TMP_FILE"
+	git checkout master
+	git fetch --all
+	git log --remotes=openzfs/master --format=%B -n 1 $OPENZFS_COMMIT > "$TMP_FILE"
+	OPENZFS_ISSUE=$(grep -oP '^[^0-9]*\K[0-9]+' -m 1 "$TMP_FILE")
+}
+
+generate_desc() {
+	USER_NAME=$(git config user.name)
+	USER_MAIL=$(git config user.email)
+	OPENZFS_COMMIT_AUTHOR=$(git log --format="%aN <%aE>" --remotes=openzfs/master -n 1 $OPENZFS_COMMIT)
+	
+	sed -i '1s/^/OpenZFS /' "$TMP_FILE"
+	sed -i 's/ / - /2' "$TMP_FILE"
+	sed -i "1 a Authored by: $OPENZFS_COMMIT_AUTHOR" "$TMP_FILE"
+	
+	echo 'Ported-by: '$USER_NAME' <'$USER_MAIL'>' >> "$TMP_FILE"
+	echo '' >> "$TMP_FILE"
+	echo 'OpenZFS-issue: https://www.illumos.org/issues/'$OPENZFS_ISSUE >> "$TMP_FILE"
+	echo 'OpenZFS-commit: https://github.com/openzfs/openzfs/commit/'$OPENZFS_COMMIT >> "$TMP_FILE"
+}
+
+#add description to commit
+add_desc_to_commit() {
+	git commit --amend -F "$TMP_FILE"
+}
+
+merge() {
+	ERR=0
+	
+	prepare_git
+	
+	echo -e "${LGREEN} - OpenZFS issue #$OPENZFS_ISSUE $OPENZFS_COMMIT ${NORMAL}"
+	echo -e "${LGREEN} Checkout new branch ${NORMAL}"
+	git branch -D "autoport-oz$OPENZFS_ISSUE"
```

Comment:
You may want to change this to `git branch -D "autoport-oz$OPENZFS_ISSUE" > /dev/null` because it displays an error if the branch doesn't exist.